qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/dictionary/impl/MultipleLangBaseDictionary.java

```diff
@@ -119,11 +119,19 @@ public long getNpredicates() {
 		return predicates.getNumberOfElements();
 	}
 
+	long cachedNobjects = -1;
+
 	@Override
 	public long getNobjects() {
-		return getNshared() + nonTyped.getNumberOfElements()
+		if (cachedNobjects != -1) {
+			return cachedNobjects;
+		}
+
+		long l = getNshared() + nonTyped.getNumberOfElements()
```

Comment:
A general question when it comes to this caching here and in BigMappedByteBuffer, what happens when the NativeStore is merged and the HDT files are rebuilt? Will we need to reset this cache or do we anyways create new objects so that it doesn't matter?

Reply:
The dictionary is bound to the HDT, so everything is static and won't change without recreating the object.